### PR TITLE
chore: Add role search param for meberships

### PR DESCRIPTION
**Pull request by AlexNti**
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🌟 New feature (non-breaking change which adds functionality)
- [ ] 🔨 Breaking change (fix or feature that would cause existing functionality)
- [ ] 📖 Docs change / refactoring / dependency upgrade to change)

 Add role as search param to filter the memberships list

### CODE REVIEW

File: tests/integration/organizations_test.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func TestOrganizations(t *testing.T) {
 	assert.Equal(t, len(organizationMembershipsWithQuery.Data), 1)
 	assert.Equal(t, organizationMembershipsWithQuery.TotalCount, int64(1))
 
-	// Should return 0 results when using the email as search param and email does not exist
+	// Should return 0 results when using the email as search param and email does not exists
```
Comment (gkats) [inline tests/integration/organizations_test.go:146]: :wrench: The comment was correct. :smile:

File: clerk/organizations.go [hunk 2]
```diff
@@ -228,6 +229,11 @@ func (s *OrganizationsService) addMembersSearchParamsToRequest(r *http.Request,
 			query.Add("user_id", userID)
 		}
 	}
+	if params.Roles != nil {
+		for _, role := range params.Roles {
+			query.Add("user_id", role)
```
Comment (gkats) [inline clerk/organizations.go:234]: :question: Is `"user_id"` intentional here? Shouldn't the parameter have to do with role?

Comment (AlexNti) [reply clerk/organizations.go]: This was a mistake. i have replace it with role.

File: clerk/organizations.go [hunk 3]
```diff
@@ -228,6 +229,11 @@ func (s *OrganizationsService) addMembersSearchParamsToRequest(r *http.Request,
 			query.Add("user_id", userID)
 		}
 	}
+	if params.Roles != nil {
```
Comment (gkats) [inline clerk/organizations.go:232]: :upside_down_face: No need to check for `nil` here. If the slice is `nil` the range won't iterate at all. 

https://go.dev/play/p/-yM16GQvnOY

**Review by gkats — APPROVED** [review]

**Review by chanioxaris — APPROVED** [review]

**Pull request merged**